### ISSUE TITLE
Managing transient fields by FieldsMetadata

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dbarra/xdocreport#227]
```
What steps will reproduce the problem?
1. create a DataModel class with transient field
2. generate fields.xml for it
3.

What is the expected output? What do you see instead?
is expected to miss transient field from fields.xml(as this field will be 
hidden in the macro). As result we have inside the transient field inside the 
fields.xml

What version of the product are you using? On what operating system?
1.0.0

Please provide any additional information below.
It will be nice to manage as keyword transient as the annotation @Transient


```

Original issue reported on code.google.com by `ruslancu...@gmail.com` on 21 Feb 2013 at 7:44

**Comment by GoogleCodeExporter** [comment]
```
I agree with you, it's a good idea.

```

Original comment by `angelo.z...@gmail.com` on 21 Feb 2013 at 8:48
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
for the moment working on patch. Test case for 225 will be modified to reach 
and this case
```

Original comment by `ruslancu...@gmail.com` on 21 Feb 2013 at 8:49

**Comment by GoogleCodeExporter** [comment]
```
Found new bug. if List is declared as: List<String> someStringList, it don't 
appear in fields list. Also, If in the dataModel tree some field is List, it is 
not declared as list in fields.xml.
```

Original comment by `ruslancu...@gmail.com` on 21 Feb 2013 at 9:29

**Comment by GoogleCodeExporter** [comment]
```
Please create a new bug please for that and if you have time please provide us 
a patch.

I think you will find a lot of problem with serializer because it manages only 
simple case.
```

Original comment by `angelo.z...@gmail.com` on 21 Feb 2013 at 9:38

**Comment by GoogleCodeExporter** [comment]
```
It will be a bit complicated to manage these bugs separately, because with this 
new bugs, it is imposible to implement new feature.
```

Original comment by `ruslancu...@gmail.com` on 21 Feb 2013 at 9:40

**Comment by GoogleCodeExporter** [comment]
```
Ok I understand, we must commit step by step your patch. I will try do that 
asap.
```

Original comment by `angelo.z...@gmail.com` on 21 Feb 2013 at 9:42

**Comment by GoogleCodeExporter** [comment]
```
ok.. don't speed up the things please, let's to be sure that these patched will 
not affect other modules.
```

Original comment by `ruslancu...@gmail.com` on 21 Feb 2013 at 9:44

**Comment by GoogleCodeExporter** [comment]
```
A question related to FieldsMetadataSerializer. For the momment serializer 
accept only Class types, and what about java primitive types? why tthese types 
are ignored? and should be primitive types be ignored?
```

Original comment by `ruslancu...@gmail.com` on 22 Feb 2013 at 8:04

**Comment by GoogleCodeExporter** [comment]
```
Here I attach a patch with solution and new junit test. 
With this patch Serializer will manage complex types.
this version of serizlizer will scan whiole tree of datamodel, and if it will 
detect some list inside, it will mark it as list. Also this patch prepare 
Abstract serializer to solve issue 180. Unfortunately solution for issue 225 
was a locally quick fix, and it doesn't permit scanning for complex data 
models, as it use vizited class types to detect cycled in tree scanning. With 
this patch, detecting of the loops is done by checking field path's, instead of 
visited class, as we can have 2 or more fields with same calss type, as we can 
have same class as parent for other.


```

Original comment by `ruslancu...@gmail.com` on 22 Feb 2013 at 8:14

Attachments:
- [issue-227-with_junit.patch](https://storage.googleapis.com/google-code-attachments/xdocreport/issue-227/comment-9/issue-227-with_junit.patch)

**Comment by GoogleCodeExporter** [comment]
```
merged in commit 679637e01c1acc6e0eacec656a9f9efcb4f50e51

Thanks for your contribution !
```

Original comment by `pascal.leclercq` on 25 Feb 2013 at 8:34
- Changed state: **Fixed**
